### ISSUE TITLE
all versions of autorest since 2.0.4245 crashing

### ISSUE THREAD

**Issue by mentat9** [Azure/autorest#3284]
I have tried multiple versions of autorest, but nothing I can find since 2.0.4245 works. They all fail like this:
```
[C:\]-> autorest
AutoRest code generation utility [version: 2.0.4283; node: v8.5.0]
(C) 2018 Microsoft Corporation.
https://aka.ms/autorest
Failure:
Error: Unable to start AutoRest Core from C:\Users\cstack\.autorest\@microsoft.azure_autorest-core@2.0.4387\node_modules\@microsoft.azure\autorest-core
Error: Unable to start AutoRest Core from C:\Users\cstack\.autorest\@microsoft.azure_autorest-core@2.0.4387\node_modules\@microsoft.azure\autorest-core
    at main (C:\Users\cstack\AppData\Roaming\npm\node_modules\autorest\dist\app.js:232:19)
    at <anonymous>

C:\Users\cstack\.autorest\@microsoft.azure_autorest-core@2.0.4387\node_modules\@microsoft.azure\autorest-core\dist\app.js:33
    autorest_core_1.Shutdown();
    ^
ReferenceError: autorest_core_1 is not defined
    at process.on (C:\Users\cstack\.autorest\@microsoft.azure_autorest-core@2.0.4387\node_modules\@microsoft.azure\autorest-core\dist\app.js:33:5)
    at emitOne (events.js:120:20)
    at process.emit (events.js:210:7)
    at process.emit (/node_modules/source-map-support/source-map-support.js:439:21)
fs.js:619
  return binding.close(fd);
                 ^

Error: EBADF: bad file descriptor, close
    at Object.fs.closeSync (fs.js:619:18)
    at StaticVolumeFile.shutdown (C:\Users\cstack\AppData\Roaming\npm\node_modules\autorest\dist\static-loader.js:352:10)
    at StaticFilesystem.shutdown (C:\Users\cstack\AppData\Roaming\npm\node_modules\autorest\dist\static-loader.js:406:17)
    at process.exit.n [as exit] (C:\Users\cstack\AppData\Roaming\npm\node_modules\autorest\dist\static-loader.js:169:11)
    at printErrorAndExit (/node_modules/source-map-support/source-map-support.js:423:11)
    at process.emit (/node_modules/source-map-support/source-map-support.js:435:16)
    at process._fatalException (bootstrap_node.js:399:26)
[C:\]->
```
I've reset, removed, uninstalled, etc. to get to a clean (single version) state and tried different versions. ```autorest --version``` works and shows the current state: (latest core version, I think):

```
[C:\]-> autorest --version
AutoRest code generation utility [version: 2.0.4283; node: v8.5.0]
(C) 2018 Microsoft Corporation.
https://aka.ms/autorest


Showing All Installed Extensions

 Type       Extension Name                           Version      Location
 core       @microsoft.azure/autorest-core           2.0.4387     C:\Users\cstack\.autorest\@microsoft.azure_autorest-core@2.0.4387


[C:\]->
```
When I uninstall autorest and install 2.0.4245, I'm able to run it and generate code, but the SDK doesn't compile with the generated code. Unsurprisingly, it appears to be out-of-date (e.g. generated code gets dropped on the old path).

Current choices are: run an old version and get wrong generation, run a newer version and be unable to generate. I'm completely blocked on this. I've been futzing with autorest for days now, and it simply won't do anything but crash. Can someone help debug this failure?

**Comment by fearthecowboy** [comment]
Update to NodeJS v10.15 LTS.

**Comment by mentat9** [comment]
Yes, that corrected the problem. Thanks for your help.
